python/sqlflow_submitter/xgboost/feature_column.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+__all__ = [
+    'numeric_column',
+    'bucketized_column',
+    'categorical_column_with_identity',
+    'categorical_column_with_vocabulary_list',
+    'categorical_column_with_hash_bucket',
+    'crossed_column',
+]
+
+
+# TODO(sneaxiy): implement faster and proper hash algorithm
+def hashing(x):
+    return hash(x)  # use builtin hash function
+
+
+def apply_transform_on_value(feature, transform_fn):
+    if len(feature) == 1:  # dense
+        return transform_fn(feature[0]),
+    else:
+        return feature[0], transform_fn(feature[1]), feature[2]
```

Comment:
Is it a sparse input? Can we add description about what each element in this tuple means.

Reply:
Done. Please see [here](https://github.com/sql-machine-learning/sqlflow/pull/2361/files#diff-2f0634b1ebc2f4375900e5a40aaad875R33).